src/gen_prog_align.jl

```diff
@@ -0,0 +1,664 @@
+    population = Array{Array, 1}(undef, population_size)
+    scores = zeros(population_size)
+    for individual = 1:population_size
+        population[individual] = Alignment(raw_data, p = p)
+        scores[individual] = objective_function(population[individual], 
+                                                score_matrix = score_matrix,
+                                                gap_opening = gap_opening,
+                                                gap_extension = gap_extension,
+                                                symbols = symbols)
+    end
+    # Variable that changes during the selection of a new generation
+    # Necessary to keep scores constant during the current generation
+    new_scores = deepcopy(scores)
+    ## Define variables necessary for the dynamic scheduling of the operations
+    # Initial chance of an operation to be selected
+    operation_chance = ProbabilityWeights(fill(1/4, 4))
+    # Initial chance of a parent to be selected
+    parent_chance = ProbabilityWeights(fill(1/population_size, population_size))
+    # Operations
+    operations = [(1, "one_point_crossover"), (2, "gap_insertion"), (3, "shifting_one_seq"), (4, "shifting_block")]
+    # Matrix to keep track of the recent performance of the operations
+    operation_performance = fill(1.0, (4, 10))
+    # Extra chances to be selected based on relative performance operations,
+    # from worst to best.
+    performance_points = [0, 1/7 * 0.5, 2/7 * 0.5, 4/7 * 0.5]
+    # Number of individuals replaced
+    n_individuals_replaced = convert(Int16, round(population_size/2))
+    # Array that keeps track for every operation which score in the operation_performance 
+    # array is the oldest, so that only one value has to be replaced.
+    renew = ones(Int, 4)
+    # Boolean variable that determines if a next generation or output will be generated.
+    go_on = true
+    # Two stop criteria 
+    n_operations_without_improvement = 0
+    n_operations = 0
+    generation = 0
+    # Every iteration in the while-loop is equal to the creation of a new generation.
+    while go_on
+        generation +=1 
+        println(join(["Maximum value generation ", string(generation), ": " ,string(maximum(scores))]))
+        # Copy of population and scores that can be changed during the creation of a new generation 
+        # while still keeping population and scores constant, making sure parents can be selected.
+        new_population = deepcopy(population)
+        new_scores = deepcopy(scores)
+        # Select the 50% individuals with the worst alignment score they will be replaced.
+        replaced_individuals = sortperm(scores)[1:n_individuals_replaced]
+        # Preallocate child which will be a candidate to be part of the next generation.
+        child = Array{String, 1}(undef, n_seq)
+        # Loop over all individuals that have to be replaced.
+        for replacement_i in replaced_individuals
+            while true
```

Comment:
It's rather hard to follow the code when the loop starts with `while true` and `break` is 100 lines below. I'd try extracting meaningful helper functions which would make it easier to read and adjust the code.